src/main/kotlin/com/noumenadigital/npl/cli/service/CloudAuthManager.kt

```diff
@@ -0,0 +1,81 @@
+package com.noumenadigital.npl.cli.service
+
+import com.noumenadigital.npl.cli.exception.CloudAuthorizationPendingException
+import com.noumenadigital.npl.cli.exception.CloudCommandException
+import com.noumenadigital.npl.cli.exception.CloudSlowDownException
+import com.noumenadigital.npl.cli.http.NoumenaCloudClient
+import com.noumenadigital.npl.cli.model.DeviceCodeResponse
+import com.noumenadigital.npl.cli.model.TokenResponse
+import com.noumenadigital.npl.cli.util.IOUtils
+import kotlinx.coroutines.delay
+import java.nio.file.Path
+import kotlin.io.path.Path
+
+class CloudAuthManager(
+    private val noumenaCloudClient: NoumenaCloudClient,
+    private val jsonFilePath: Path = Path(System.getProperty("user.home"), ".noumena", "noumena.yaml"),
+) {
+    suspend fun login(output: ColorWriter) {
+        val deviceCode = noumenaCloudClient.requestDeviceCode()
+        openBrowser(deviceCode.verificationUri, output)
+        output.info("Please use the following code to complete authentication in your browser: ${deviceCode.userCode}")
```

Comment:
```suggestion
        openBrowser(deviceCode.verificationUriComplete, output)
        output.info("Please open the following URL in your browser: ${deviceCode.verificationUriComplete}")
```

this way the user doesn't have to copy-paste the code

Reply:
I like your idea